backend/fms_core/utils.py

```diff
@@ -163,3 +165,15 @@ def make_generator(obj: Union[Any, None, Iterable[Any]]) -> Generator[Any, None,
                 yield x
         except TypeError:
             yield obj
+
+def make_timestamped_filename(file_name: str, extension: str) -> str:
+    """
+    Creates a file name composed of a base file name followed by a timestamp, followed
+    by an extension, eg "MyFile_2022-11-25_08-13-45.json".
+
+    The file name and timestamp are separated by an underscore.
+    The extension parameter should not include a '.', as one is inserted automatically.
```

Comment:
You might be better to align with os.path.splitext which return the "." as part of the extension. If we ever need to add a timestamp to an existing filename + extension this will be easier.

Reply:
Ah yes. I had forgotten about `splitext`.